main.go

```diff
@@ -0,0 +1,120 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"sort"
+)
+
+type Entry struct {
+	ID          uint
+	FirstName   string
+	LastName    string
+	PhoneNumber string
+}
+
+func mainMenu() (choice int) {
+	fmt.Println(`Choose your action:
+1) List all entries.
+2) Add new entry.
+3) Remove an entry by ID.
+4) Quit`)
+	fmt.Scanln(&choice)
+	return choice
+}
+
+func openFile() []Entry {
+	var entries []Entry
+	file, err := os.Open("phonebook.json")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer file.Close()
+	if err := json.NewDecoder(bufio.NewReader(file)).Decode(&entries); err != nil {
+		log.Fatal(err)
+	}
+	return entries
+}
+
+func listEntries() {
+	entries := openFile()
+	sort.Slice(entries, func(i, j int) bool { return entries[i].LastName < entries[j].LastName })
+	fmt.Printf("%10s %20s %20s %20s\n", "ID", "Last Name", "First Name", "Phone Number")
+	fmt.Println("-------------------------------------------------------------------------")
+	for i, e := range entries {
+		fmt.Printf("%10d %20s %20s %20s\n", e.ID, e.LastName, e.FirstName, e.PhoneNumber)
+		if (i+1)%20 == 0 && i < len(entries)-1 {
+			fmt.Print("Press <ENTER> to continue...")
+			fmt.Scanln()
+			fmt.Printf("%10s %20s %20s %20s\n", "ID", "Last Name", "First Name", "Phone Number")
+			fmt.Println("-------------------------------------------------------------------------")
+		}
+	}
+}
+
+func addNew() {
+	var ID uint
+	entries := openFile()
+	for i, e := range entries {
+		if i == len(entries)-1 {
+			ID = e.ID + 1
+		}
+	}
+	entry := Entry{ID: ID}
+	fmt.Print("Enter last name:")
+	fmt.Scan(&entry.LastName)
+	fmt.Print("Enter first name:")
+	fmt.Scan(&entry.FirstName)
+	fmt.Print("Enter phone number:")
+	fmt.Scan(&entry.PhoneNumber)
+	entries = append(entries, entry)
+	file, err := os.Create("phonebook.json")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer file.Close()
+	if err := json.NewEncoder(file).Encode(entries); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func removeByID() {
+	entries := openFile()
+	var ID uint
+	fmt.Print("Enter ID:")
+	fmt.Scan(&ID)
+	for i, e := range entries {
+		if ID == e.ID {
+			entries = append(entries[:i], entries[i+1:]...)
+		}
+	}
+	file, err := os.Create("phonebook.json")
```

Comment:
nit: Would be cool if you factored savePhonebook in the same way you did with `openFile`
just to make it more readable